### PR TITLE
lsp-completion: check falsy value for some completion members

### LINKED ISSUE #2761: Unable to filter completions when `filterText` is an empty string
**Describe the bug**
The LSP spec states that when a `CompletionItem`'s `filterText` is falsy, it should use the `label` to perform filtering. Recently, zls switched to returning an empty string for `filterText`, following the LSP spec (https://github.com/zigtools/zls/pull/251, https://github.com/microsoft/language-server-protocol/issues/1222). However, with this change, it appears that typing any characters to to filter down the list of completions using company causes all the completions to disappear, which I expect is because they are filtered out given that `filterText` is `""`.

**To Reproduce**
Install [zls](https://github.com/zigtools/zls) and open any zig file and try completions. After entering a '.' and triggering completions, company displays a list of completions. Any text following that causes completions to be filtered away.

**Expected behavior**
`filterText` should not be used for filtering when it is the empty string.

**Which Language Server did you use**
zls: https://github.com/zigtools/zls

**OS**
NixOS 21.05

### PR DESCRIPTION
**Pull request by kiennq**
Fix #2761

### CODE REVIEW

File: lsp-completion.el [hunk 1]
```diff
@@ -194,7 +198,7 @@ Return `nil' when fails to guess prefix."
     (lsp--position-to-point (lsp:range-start (lsp:text-edit-range text-edit?))))
    (t
     (-let* (((&CompletionItem :label :insert-text?) item)
-            (text (or insert-text? label))
+            (text (or (unless (lsp-falsy? insert-text?) insert-text?) label))
```
Comment (yyoncho) [inline lsp-completion.el:201]: this could be rewritten as `(if (lsp-falsy...`, here and on a few more places.

**Review by yyoncho — APPROVED** [review]
Looks good, one minor nit:

File: lsp-completion.el (hunk 1 above)
Comment (kiennq) [reply lsp-completion.el:201]: Yeah, I'm purposely writing like this so we can add more term to the `or` clause` if the LSP protocol ever changes/adds new text. Do you think I can merge keep this?

Comment (yyoncho) [reply lsp-completion.el:201]: ok, it is not a big deal...

**Pull request merged**